march_hardware/include/march_hardware/imotioncube/motor_controller_state.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Project March.
+#ifndef MARCH_HARDWARE_MOTOR_CONTROLLER_STATE_H
+#define MARCH_HARDWARE_MOTOR_CONTROLLER_STATE_H
+
+#include <string>
+#include "imotioncube_state.h"
+
+namespace march
+{
+struct MotorControllerState
+{
+public:
+  MotorControllerState() = default;
+
+  float motorCurrent;
+  float controllerVoltage;
+  float motorVoltage;
+  int absoluteEncoderValue;
+  int incrementalEncoderValue;
+  double absoluteVelocity;
+  double incrementalVelocity;
+
+  std::string statusWord;
+  std::string motionError;
+  std::string detailedError;
+  std::string secondDetailedError;
+  IMCState state;
+  std::string detailedErrorDescription;
+  std::string motionErrorDescription;
+  std::string secondDetailedErrorDescription;
```

Comment:
These are imc specific, so shouldn't they be moved to the imc_state.

Reply:
imc_state is basically just an enumeration of "Not Ready to swtich on" and alike states. Hence this is not suitable to move these to that object. The naming is confusing so I improved it slightly by renaming MotorControllerState to MotorControllerStates

Reply:
I still agree with Rutger here, these variables are _very_ IMc specific. You could create Error class where every motor controller could implement its own representation of an error. And then add a method like `getErrorDescription` which returns a universal error description.

Reply:
Nice I like that suggestion :+1:. As the checkState method basically already implemented such a universal error description, I moved that to a new method getErrorStatus instead of implementing a separate class for this. Let me know what you think.